### DISCUSSION TITLE
How to make push notification when  my mobile is offline?

### DISCUSSION THREAD

**Discussion by fwlxpro** [magicbell/community#201] [Questions] [answered]
Hi i need for my corporation when user be offline he have still push notifications

**Comment by unamashana** [comment]
Hey @Snefcz apologies for missing this. The delivery here depends on the platform (iOS/Android). If they support redelivery when a user comes online, and you need to send a field in the API to make this happen, you can look at provider specific overrides - https://www.magicbell.com/docs/notification-rendering/overrides

**Comment by stigi** [comment, marked as answer]
Let me add some more context. What OP is probably looking for is known as [_Local Notifications_ on iOS](https://developer.apple.com/documentation/usernotifications/scheduling-a-notification-locally-from-your-app). A similar mechanism exists [on Android](https://developer.android.com/develop/ui/views/notifications/build-notification).

Those notifications appear on the users lock screen, similar to remote push notifications. They are triggered locally from the users device though, based on various possible events (could be a timer, a significant location change, a background task) and don't require network connections.

Those notifications can be used in parallel to MagicBell issued push notifications, but are outside of the MagicBell domain and need custom implementation in your app.
